google/resource_bigtable_gc_policy.go

```diff
@@ -0,0 +1,227 @@
+package google
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"cloud.google.com/go/bigtable"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+const (
+	GCPolicyModeIntersection = "INTERSECTION"
+	GCPolicyModeUnion        = "UNION"
+)
+
+func resourceBigtableGCPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceBigtableGCPolicyCreate,
+		Read:   resourceBigtableGCPolicyRead,
+		Delete: resourceBigtableGCPolicyDestroy,
+
+		Schema: map[string]*schema.Schema{
+			"instance_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"table": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"family": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"mode": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{GCPolicyModeIntersection, GCPolicyModeUnion}, false),
+			},
+
+			"max_age": {
+				Type:     schema.TypeMap,
```

Comment:
```suggestion
				Type:     schema.TypeList,
```

Reply:
Types in Terraform are a little unintuitive, and it has no real way to represent nested objects. `TypeList` is the best approximate.